EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -47,344 +55,89 @@ void resetSenseArray(uint32_t senses[]);
             {
-                uint8_t contactorError = 0;
-
-                const uint8_t commonSense = !HAL_GPIO_ReadPin(COMMON_SENSE_GPIO_Port, COMMON_SENSE_Pin);
-                const uint8_t chargeSense = !HAL_GPIO_ReadPin(CHARGE_SENSE_GPIO_Port, CHARGE_SENSE_Pin);
-                const uint8_t dischargeSense = !HAL_GPIO_ReadPin(DISCHARGE_SENSE_GPIO_Port, DISCHARGE_SENSE_Pin);
-
-                if (!commonSense || (hasChargeBeenSet && !chargeSense && !orionStatus.chargeContactorOverriden))
-                {
-                    disconnectContactors(1);
-                    state = CONTACTOR_DISCONNECTED;
-                    break;
-                }
-
-                if (!dischargeSense)
-                {
-                    int numContactors = chargeSense ? 3 : 2;
-                    hasDischargeBeenSet = isContactorSet(DISCHARGE_SENSE_Pin, DISCHARGE_SENSE_GPIO_Port, numContactors);
-
-                    if (hasDischargeBeenSet)
-                    {
-                        // Enable high voltage
-                        HAL_GPIO_WritePin(HV_ENABLE_GPIO_Port, HV_ENABLE_Pin, GPIO_PIN_SET);
-                    }
-                    else // Turn back off if not set
-                    {
-                        HAL_GPIO_WritePin(DISCHARGE_ENABLE_GPIO_Port, DISCHARGE_ENABLE_Pin, GPIO_PIN_RESET);
-                    }
-                }
-
-                switch (prevState)
-                {
-                    case DONE:
-                        contactorError = !hasDischargeBeenSet;
-                        break;
-
-                    case CHARGE_ENABLE_CHECK:
-                        contactorError = !(hasChargeBeenSet && hasDischargeBeenSet);
-                        break;
-
-                    default: // Something went really wrong in the logic, stop everything
-                        disconnectContactors(0);
-                        state = SHUTDOWN;
-                        continue;
-                }
-
-                if (osMutexWait(auxStatus.auxStatusMutex, 0) != osOK)
-                {
-                    continue;
-                }
-
-                auxStatus.contactorError = contactorError;
-
-                if (hasDischargeBeenSet && hasChargeBeenSet)
-                {
-                    auxStatus.startUpSequenceDone = 1;
-                    state = DONE;
-                }
-
-                osMutexRelease(auxStatus.auxStatusMutex);
-
-                if (!hasChargeBeenSet)
-                {
-                    prevState = state;
-                    state = CHARGE_ENABLE_CHECK;
-                    HAL_GPIO_WritePin(CHARGE_ENABLE_GPIO_Port, CHARGE_ENABLE_Pin, GPIO_PIN_SET);
-                }
-                else if (!hasDischargeBeenSet)
-                {
-                    prevState = state;
-                    state = CHARGE_ENABLE_CHECK;
-                }
-
-
-
-                break;
+                dischargeEnableCheck(&toAlsoBeNamed);
             }
 
             case DONE:
```

Comment:
I guess this is a good time to rename this state, `DONE` made sense back then, but now it doesn't represent any real finality in the Aux BMS. What kind of name should we give this?

Reply:
Maybe `IDLE` since this is a state that the Aux BMS waits around for things to change?